src/components/FormInput.jsx

```diff
@@ -0,0 +1,117 @@
+    const [openSuggestedLocations, setOpenSuggestedLocations] = useState(false)
+    const cityName = locationRef.current?.value
+
+    const onsearchHandler = () => {
+        if (cityName !== '') {
+            const searchObj = { cityName: locationRef.current?.value }
+            getSearchData(searchObj).then((res) => {
+                setSearchData(res)
+                setLoading(false)
+            })
+        } else {
+            return
+        }
+    }
+    const getSuggestedLocations = () => {
+        if (cityName !== '') {
+            setOpenSuggestedLocations(false)
+            getLocation(cityName).then((res) => {
+                setSuggestedLocations(res)
+            }).catch(() => {
+                setSearchData([])
+            })
+        } else {
+            return
+        }
+    }
+
+    useEffect(() => {
+        if (searchData['length'] !== 0 && suggestedLocations !== '') {
+            navigateTo("/map", { state: { searchData, suggestedLocations } })
+        }
+    }, [searchData, suggestedLocations])
+
+    const suggesteKeyWordToggle = () => {
+        if (openSuggestedLocations) {
+            setOpenSuggestedLocations(false)
+        } else {
+            setOpenSuggestedLocations(true)
+        }
+    }
+
+    const suggestedLocationsList = SUGGESTIONS.map((el) => {
+        const moveToSuggestedCityHandler = () => {
+            setOpenSuggestedLocations(false)
+            const searchObj = { cityName: el.capital }
+            setLoading(true)
+            getSearchData(searchObj).then((res) => {
+                setSearchData(res)
+                setLoading(false)
+            })
+            getLocation(el.capital).then((res) => {
+                setSuggestedLocations(res)
+            }).catch(() => {
+                setSearchData([])
+            })
+        }
+
+        return (
+            <li key={el.id} onClick={moveToSuggestedCityHandler}>
+                <Discovery className={classes['location__discovery-icon']} />
+                <span className={classes['location__suggested']}>{el.location}</span>
+            </li>
+        )
+    })
+
+    return (
+        <section className={classes['form']}>
+            <div className={classes['location']}>
+                <form className={classes['location__form']}>
+                    <div onClick={() => { onsearchHandler(); getSuggestedLocations() }}><Search className={classes['location__search-icon']} /></div>
+                    <input
+                        type="text"
+                        placeholder="Search for location..."
+                        className={classes['location__input']}
+                        ref={locationRef}
+                        onClick={suggesteKeyWordToggle}
+                        onKeyDown={(e) => {
+                            if (e.key == 'Enter') {
+                                e.preventDefault()
+                                onsearchHandler(); getSuggestedLocations()
```

Comment:
Can this be inside onSearchHandler or combined into one function if both of them are always called together? (didn't check all the usage, not sure if it's the case)